gcloud/monitoring/group.py

```diff
@@ -0,0 +1,507 @@
+    projects/            # static prefix
+    (?P<project>[^/]+)   # initial letter, wordchars + hyphen
+    /groups/             # static midfix
+    (?P<name>[^/]+)      # initial letter, wordchars + allowed punc
+""", re.VERBOSE)
+
+
+def group_id_from_name(path, project=None):
+    """Validate a group URI path and get the group ID.
+
+    :type path: string
+    :param path: URI path for a group API request.
+
+    :type project: string or None
+    :param project: The project associated with the request. It is
+                    included for validation purposes.
+
+    :rtype: string
+    :returns: Group ID parsed from ``path``.
+    :raises: :class:`ValueError` if the ``path`` is ill-formed or if
+             the project from the ``path`` does not agree with the
+             ``project`` passed in.
+    """
+    if not path:
+        return ''
+    return _name_from_project_path(path, project, _GROUP_TEMPLATE)
+
+
+def group_name_from_id(project, group_id):
+    """Build the group name given the project and group ID.
+
+    :type project: string
+    :param project: The project associated with the group.
+
+    :type group_id: string
+    :param group_id: The group ID.
+
+    :rtype: string
+    :returns: The fully qualified name of the group.
+    """
+    return 'projects/{project}/groups/{group_id}'.format(
+        project=project, group_id=group_id)
+
+
+class Group(object):
+    """A dynamic collection of monitored resources.
+
+    :type client: :class:`gcloud.monitoring.client.Client`
+    :param client: A client for operating on the metric descriptor.
+
+    :type group_id: string or None
+    :param group_id: The ID of the group.
+
+    :type name: string or None
+    :param name:
+          The fully qualified name of the group in the format
+          "projects/<project>/groups/<id>". This is a read-only property.
+
+    :type display_name: string
+    :param display_name:
+        A user-assigned name for this group, used only for display purposes.
+
+    :type parent_name: string
+    :param parent_name:
+        The fully qualified name of the group's parent, if it has one.
+
+    :type filter_string: string
+    :param filter_string:
+        The filter string used to determine which monitored resources belong to
+        this group.
+
+    :type is_cluster: boolean
+    :param is_cluster:
+        If true, the members of this group are considered to be a cluster. The
+        system can perform additional analysis on groups that are clusters.
+
+    :raises: :exc:`ValueError` if both ``group_id`` and ``name`` are specified.
+    """
+    def __init__(self, client, group_id=None, name=None, display_name='',
+                 parent_name='', filter_string='', is_cluster=False):
+        if not (group_id is None or name is None):
+            raise ValueError('At most one of "group_id" and "name" can be '
+                             'specified')
+
+        self.client = client
+        if group_id is not None:
```

Comment:
Not required anymore - I updated the init to only take in the group_id. The name and parent_name are constructed from the respective IDs.